Shared/Theme/ThemeManager.swift

```diff
@@ -63,41 +66,37 @@ final class DefaultThemeManager: ThemeManager, Notifiable {
 
     // MARK: - ThemeManager
 
-    func getInterfaceStyle() -> UIUserInterfaceStyle {
+    public func getInterfaceStyle() -> UIUserInterfaceStyle {
         return currentTheme.type.getInterfaceStyle()
     }
 
-    func changeCurrentTheme(_ newTheme: ThemeType) {
+    public func changeCurrentTheme(_ newTheme: ThemeType) {
         guard currentTheme.type != newTheme else { return }
         currentTheme = newThemeForType(newTheme)
 
         // overwrite the user interface style on the window attached to our scene
         // once we have multiple scenes we need to update all of them
-        UIWindow.keyWindow?.overrideUserInterfaceStyle = currentTheme.type.getInterfaceStyle()
-
+        if #available(iOS 15, *) {
+            appDelegate?.window??.windowScene?.keyWindow?.overrideUserInterfaceStyle = currentTheme.type.getInterfaceStyle()
+        }
         // old way of updating the theme. We should pass in and save sceneDelegate instead of appDelegate
         appDelegate?.window??.overrideUserInterfaceStyle = currentTheme.type.getInterfaceStyle()
```

Comment:
Should this be in an else? Or is it fine this way?

Reply:
It's fine to call both. I will clean this part up when we move to multi-window support.